bundles/org.eclipse.equinox.http.servlet/src/org/eclipse/equinox/http/servlet/internal/context/ProxyContext.java

```diff
@@ -49,36 +52,64 @@ public void destroy() {
 			deleteDirectory(proxyContextTempDir);
 	}
 
-	public synchronized String getServletPath() {
-		return (servletPath == null ? Const.BLANK : servletPath);
+	public String getServletPath() {
+		return Const.BLANK;
 	}
 
-	public synchronized void createContextAttributes(
+	public void createContextAttributes(
 		ContextController controller) {
 
-		ContextAttributes attributes = attributesMap.get(controller);
-		if (attributes == null) {
-			attributes = new ContextAttributes(controller);
-			attributesMap.put(controller, attributes);
-		}
-		attributes.addReference();
+		attributesMap.compute(
+			controller,
+			new BiFunction<ContextController, ContextAttributes, ContextAttributes>() {
```

Comment:
We can't use this since it's Java 8 and this branch is built using Java 6. Would you mind changing this to a java 6 construct please?